src/app/chip-zcl-zpro/command-encoder/decoder.c

```diff
@@ -26,73 +26,73 @@
 #include <stdio.h>
 #include <string.h>
 
-// TODO: Make this return a CHIP_ERROR
-bool extractApsFrame(void * buffer, uint32_t buf_length, EmberApsFrame * outApsFrame)
+uint16_t extractApsFrame(void * buffer, uint32_t buf_length, EmberApsFrame * outApsFrame)
 {
+
     if (buffer == NULL || buf_length == 0 || outApsFrame == NULL)
     {
-        return false;
+        return 0;
     }
     // Skip first byte, because that's the always-0 frame control.
     uint8_t nextByteToRead = 1;
 
     if (nextByteToRead >= buf_length)
     {
-        return false;
+        return 0;
     }
     memcpy(&outApsFrame->profileId, buffer + nextByteToRead, sizeof(outApsFrame->profileId));
```

Comment:
is there an issue filed that this code won't work on a BIGENDIAN machine?

Reply:
Yes, https://github.com/project-chip/connectedhomeip/issues/1632